delivery_send_to_shipper_at_operation/models/stock_picking.py

```diff
@@ -0,0 +1,85 @@
+# Copyright 2021 Camptocamp SA
+# License AGPL-3.0 or later (http://www.gnu.org/licenses/agpl)
+
+from odoo import fields, models
+
+
+class StockPicking(models.Model):
+    _inherit = "stock.picking"
+
+    delivery_notification_sent = fields.Boolean(default=False)
+
+    def _send_confirmation_email(self):
+        for picking in self:
+            if not picking.carrier_id:
+                # If the current operation has no carrier defined, but a carrier
+                # has been found in one of the chained transfers and is
+                # configured to match the current operation type: force the
+                # sending of the delivery notice to the carrier
+                carrier = picking.get_carrier_from_chained_picking()
+                related_ship = picking._get_carrier_operation_from_chained_picking()
+                if (
+                    carrier.integration_level == "rate_and_ship"
+                    and carrier.send_delivery_notice_on == "custom"
+                    and picking.picking_type_id
+                    in carrier.send_delivery_notice_picking_type_ids
+                ):
+                    picking.carrier_id = carrier
+                    picking.send_to_shipper()
+                    picking.carrier_id = False
```

Comment:
wouldn't make sense to keep the carrier here?

Reply:
I preferred to unset it but good question, now the pack is done, we could keep it for traceability, especially if for whatever reason we change the carrier later on the ship (maybe something to consider too: unset the flag on the ship if we change its carrier).

Reply:
I remove the line, now the carrier is kept on the operation.